java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -333,28 +334,28 @@ JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_run(JNIEnv* jniEnv
   jobjectArray outputArray = NULL;
 
   // Create the buffers for the Java input & output strings, and the input pointers
-  const char** inputNames = malloc(sizeof(char*) * numInputs);
```

Comment:
The old code forgets to deal with integer overflow.  The result of sizeof(char*) * numInputs   could be bigger than the max value of `size_t`. The new old code has safety check for this.